### PR TITLE
Add test most recent retrievale root. Closes filcdn/roadmap#4

### PR DESCRIPTION
**Pull request by juliangruber**
Closes filcdn/roadmap#4

I decided to reduce scope and simply add a second loop that tests retrieval of the most recent retrievable root. If it isn't retrievable, it will create an #alert. We can iterate from there if needed.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -354,3 +354,114 @@ async function pickRandomFileWithCDN({
...
+ * }>}
+ *   The CommP CID of the file.
+ */
+async function getMostRecentFileWithCDN({
+  pdpVerifier,
+  pandoraService
+}) {
+  for (
+    let proofSetId = (await pdpVerifier.getNextProofSetId()) - 1n;
+    proofSetId >= 0n;
```
Comment (bajtos) [inline index.js:406]: We should stop at `FROM_PROOFSET_ID`. ProofSet older than that may not work, and the original loop never tests them.

Since we are very unlikely to encounter this edge case, I don't mind keeping your current version that iterates all the way down to the proof set id 0.

**Review by bajtos — COMMENTED** [review]
LGTM, but please fix the CI. (`npm run lint:fix`?)

**Comment by juliangruber** [comment]
> LGTM, but please fix the CI. (`npm run lint:fix`?)

b22e0a1

File: index.js (hunk 1 above)
Comment (juliangruber) [reply index.js]: 2810173

Comment (juliangruber) [reply index.js]: I decided to add it for symmetry and to prevent surprises

File: index.js [hunk 2]
```diff
@@ -354,3 +354,116 @@ async function pickRandomFileWithCDN({
...
+ * @param {BigInt} args.FROM_PROOFSET_ID
+ * @returns {Promise<{
+ *   rootCid: string
+ *   proofSetId: BigInt
+ *   rootId: BigInt
+ *   clientAddress: string
+ * }>}
+ *   The CommP CID of the file.
+ */
+async function getMostRecentFileWithCDN({
```
[the remark this answers is not in the document]
Comment (juliangruber) [reply index.js:403]: This was for consistency within the file, there's also `pickRandomFileWithCDN`
